source/Camera/InfoUtilOther/ZividBenchmark/ZividBenchmark.cpp

```diff
@@ -0,0 +1,473 @@
+                                            - computeMedianDuration(captureDuration));
+        filterProcessingDurations.push_back(computeAverageDuration(captureDurationReflection)
+                                            - computeAverageDuration(captureDuration));
+        filterProcessingDurations.push_back(computeMedianDuration(captureDurationGaussianReflection)
+                                            - computeMedianDuration(captureDuration));
+        filterProcessingDurations.push_back(computeAverageDuration(captureDurationGaussianReflection)
+                                            - computeAverageDuration(captureDuration));
+        printFilterResults(filterProcessingDurations);
+    }
+
+    Zivid::Settings2D makeSettings2D(const size_t exposureTime)
+    {
+        Zivid::Settings2D settings;
+        settings.set(Zivid::Settings2D::ExposureTime(std::chrono::microseconds{ exposureTime }));
+        return settings;
+    }
+
+    void benchmarkCapture2D(Zivid::Camera &camera, const Zivid::Settings2D &settings, const size_t numFrames)
+    {
+        printCapture2DHeader(numFrames, settings);
+
+        for(size_t i = 0; i < 5; i++) // Warmup frames
+        {
+            camera.capture2D(settings);
+        }
+
+        std::vector<Duration> captureDurations;
+        std::vector<std::vector<Duration>> durations;
+        for(size_t i = 0; i < numFrames; i++)
+        {
+            const auto beforeCapture = HighResClock::now();
+            const auto frame2D = camera.capture2D(settings);
+            const auto afterCapture = HighResClock::now();
+
+            captureDurations.push_back(afterCapture - beforeCapture);
+        }
+        durations.push_back(captureDurations);
+        printCapture2DResults(durations);
+    }
+
+    void benchmarkSave(Zivid::Camera &camera, const size_t numFrames)
+    {
+        printSaveHeader(numFrames);
+
+        auto frame = camera.capture();
+        frame.getPointCloud();
+
+        std::vector<std::vector<Duration>> saveDurations;
+        std::vector<std::string> fileNames{ "Zivid3D.zdf", "Zivid3D.ply", "Zivid3D.pcd", "Zivid3D.xyz" };
+        for(const auto &fileName : fileNames)
+        {
+            std::vector<Duration> durationsPerFormat;
+            for(size_t j = 0; j < numFrames; j++)
+            {
+                const auto beforeSave = HighResClock::now();
+                frame.save(fileName);
+                const auto afterSave = HighResClock::now();
+
+                durationsPerFormat.push_back(afterSave - beforeSave);
+            }
+            saveDurations.push_back(durationsPerFormat);
+        }
+        printSaveResults(saveDurations);
+    }
+} // namespace
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        printZividInfo(zivid);
+        printPrimarySeparationLine();
+        printCentered("Starting Zivid Benchmark");
+
+        const size_t numConnects = 10;
```

Comment:
Could it be an idea to gather all the constant values (configuration parameters) at the top of main()? Or don't we ever want anyone to modify those? 
A couple of other alternatives would be to allow them as CLI args (with defaults) or a .yaml?

Reply:
Yes that is a good idea, same with the columnWidt parameter I proposed.

Reply:
Done